src/random.jl

```diff
@@ -83,20 +83,30 @@ desired connectance value. If the simulated network has a tolerance x, the
 target connectance is c, then the relative error is |1-x/c|.
 
 """
-function nichemodel(S::Int64, C::Float64; tolerance::Float64=0.05, toltype::Symbol=:abs)
+function nichemodel(S::Int64, C::Float64; tolerance::Float64 = 0.05, toltype::Symbol = :abs)
     @assert C < 1.0
     @assert C > 0.0
     @assert tolerance > 0.0
     @assert toltype ∈ [:abs, :rel]
     L = round(Int64, C * S^2)
     A = nichemodel(S, L)
     if toltype == :abs
-      tolfunc = (x) -> abs(x-C) < tolerance
+        tolfunc = (x) -> abs(x - C) < tolerance
     else
-      tolfunc = (x) -> abs(1-x/C) < tolerance
+        tolfunc = (x) -> abs(1 - x / C) < tolerance
     end
     while !(tolfunc(connectance(A)))
       A = nichemodel(S, L)
     end
     return A
 end
+
+"""
+Return BitVector of primary producers
+1 - The species is a primary producer.
+0 - The species is a consumer (i.e. consumes at least one other species).
+"""
+function whoisproducer(A)
+    isproducer = vec(sum(A, dims = 2) .== 0)
```

Comment:
`isproducer` contains boolean values, so.. is `A` also supposed to contain boolean values? If yes, then try
```
isproducer = any(A, dims = 2)
```
instead ;)